### ISSUE TITLE
Change the captain root domain

### ISSUE THREAD

**Issue by danijel-orsolic** [githubsaturn/captainduckduck#252]
Hello,

I'm wondering how can I change the captain root domain name, without using the GUI. The IP and server is the same.

Thank you!

**Comment by githubsaturn** [comment]
You can directly modify the configuration file located at `/captain/config.conf` and then run `docker service update captain-captain --force`

But please note that if you have SSL (HTTPS) enabled, you cannot just update your domain and hope everything works. That's why, after enabling SSL, the root domain name is changed to read-only on the web dashboard. There should be very little benefit from being able to change your root domain name as you can always attach as many custom domains as you want to your apps.

To sum up:
- Before enabling SSL: just use the web dashboard or directly modify the config file as explained above.
- After enabling SSL: STRONGLY DISCOURAGED. But if you really have to do so, manually edit config file, disable SSL for root domain and all apps, disable force to HTTPS everywhere in the config file and then run `docker service update captain-captain --force`

**Comment by drmrbrewer** [comment]
> That's why, after enabling SSL, the root domain name is changed to read-only on the web dashboard

![image](https://user-images.githubusercontent.com/9009792/65370260-b1cbe400-dc4e-11e9-913f-6fe30438f255.png)

I'm a bit confused.  It seems possible for me to change the root domain in my dashboard (see above)?   Not that I'm want to test it, because it is likely to mess everything up.  But I'm considering retiring the domain on which caprover is hosted, so would like to work out a strategy for doing this (both caprover and all my apps are sitting under this domain, all with https enabled).

**Comment by githubsaturn** [comment]
The feature was added in 1.3: https://github.com/caprover/caprover/blob/master/CHANGELOG.md#130---2019-03-04

> Ability to change the root domain post installation

This feature is now supported, however, it's not recommended in production apps, as any potential errors during renaming might cause SSL issues and etc.

**Comment by drmrbrewer** [comment]
Will that change it in all the apps too?  i.e. `https://myapp.myserver.mydomain.com` --> `https://myapp.mynewserver.mynewdomain.com`?

**Comment by githubsaturn** [comment]
Yes

**Comment by drmrbrewer** [comment]
What are the potential risks?  Of course from our side we need to ensure that the DNS entries for the new domain are all in place, but from the CapRover side of things what could go wrong?

**Comment by githubsaturn** [comment]
Nothing specific. But theoretically, it can leave the config in a bad state. If you have a backup of the config file, it should be fine.
